Makefile.am

```diff
@@ -298,6 +297,18 @@ install-exec-hook:
 			cd ${DESTDIR}${libdir} && ln -f -s lib@MPICXXLIBNAME@@SHLIB_EXT@ libmpichcxx@SHLIB_EXT@ ; \
 		fi ; \
 	fi
+if BUILD_ABI_LIB
+	if test -e ${DESTDIR}${bindir}/@MPICC_NAME@ ; then \
+		if test "@MPICC_NAME@" != "@MPICC_ABI_NAME@" ; then \
+			cd ${DESTDIR}${bindir} && ln -f -s @MPICC_NAME@ @MPICC_ABI_NAME@ ; \
+		fi ; \
+	fi ; \
+	if test -e ${DESTDIR}${bindir}/@MPICXX_NAME@ ; then \
+		if test "@MPICXX_NAME@" != "@MPICXX_ABI_NAME@" ; then \
+			cd ${DESTDIR}${bindir} && ln -f -s @MPICXX_NAME@ @MPICXX_ABI_NAME@ ; \
+		fi ; \
+	fi ;
+endif BUILD_ABI_LIB
```

Comment:
I am surprised that this works :)

Reply:
Why? I makes perfect sense! However note that this rule eventually uses multiple shell invocations, and not just one.

PS: I guess now you understand what I was struggling with.